### PR TITLE
Disable the exllama on all non-cuda devices.

### PR DESCRIPTION
**Pull request by yuanwu2017**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by yao-matrix** [comment]
@IlyasMoutawwakil , could you help review? This PR is to disable exllama on non-cuda devices, since other devices don't support it yet, this will enable devices like intel habana can work w/ autogptq. The one open is: do we need add rocm into SUPPORT_EXLLAMA_DEVICES, since the exllama says "it's theoretically supported but not validated", could you share your insights?

**Comment by IlyasMoutawwakil** [comment]
@yao-matrix rocm is registered as cuda so that shouldn't be an issue, I think there's no need for a list of supported devices as exllama is written in cuda and will probably on work on cuda/hip/rocm machines, that should simplify the PR.

**Review by SunMarc — APPROVED** [review]
Thanks for the PR @yuanwu2017  ! LGTM !

File: optimum/gptq/quantizer.py [hunk 1]
```diff
@@ -546,7 +546,9 @@ def tmp(_, input, output):
 
         if self.bits == 4:
             # device not on gpu
-            if device == torch.device("cpu") or (has_device_map and any(d in devices for d in ["cpu", "disk"])):
+            if device.type not in SUPPORT_EXLLAMA_DEVICES or (
```
Comment (IlyasMoutawwakil) [inline optimum/gptq/quantizer.py:549]:
```suggestion
            if device.type != "cuda" or (
```

File: optimum/gptq/quantizer.py [hunk 2]
```diff
@@ -589,13 +591,14 @@ def post_init_model(self, model):
                 The input model
         """
         if self.bits == 4 and not self.disable_exllama:
-            if get_device(model) == torch.device("cpu") or (
-                hasattr(model, "hf_device_map") and any(d in model.hf_device_map for d in ["cpu", "disk"])
+            if get_device(model).type not in SUPPORT_EXLLAMA_DEVICES or (
```
Comment (IlyasMoutawwakil) [inline optimum/gptq/quantizer.py:594]:
```suggestion
            if get_device(model).type != "cuda" or (
```

File: optimum/gptq/quantizer.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
 from ..utils import is_accelerate_available, is_auto_gptq_available
 from ..utils.modeling_utils import recurse_getattr
-from .constants import GPTQ_CONFIG
+from .constants import GPTQ_CONFIG, SUPPORT_EXLLAMA_DEVICES
```
Comment (IlyasMoutawwakil) [inline optimum/gptq/quantizer.py:30]:
```suggestion
from .constants import GPTQ_CONFIG
```

File: optimum/gptq/constants.py [hunk 4]
```diff
@@ -21,3 +21,4 @@
 ]
 
 GPTQ_CONFIG = "quantize_config.json"
+SUPPORT_EXLLAMA_DEVICES = ["cuda"]
```
Comment (IlyasMoutawwakil) [inline optimum/gptq/constants.py:24]:
```suggestion
```

**Comment by IlyasMoutawwakil** [comment]
@yao-matrix can you run `make style` please

**Comment by yuanwu2017** [comment]
@IlyasMoutawwakil The error is not related with the patch. Please help to merge it.

**Pull request merged**